pysoa/client/client.py

```diff
@@ -539,96 +542,112 @@ def call_jobs_parallel(
             responses.append(response)
 
         if expansions:
-            self._perform_expansion(actions_to_expand, expansions)
+            kwargs.pop('continue_on_error', None)
+            self._perform_expansion(actions_to_expand, expansions, **kwargs)
 
         return responses
 
-    def _perform_expansion(self, actions, expansions):
+    def _perform_expansion(self, actions, expansions, **kwargs):
         # Perform expansions
         if expansions and hasattr(self, 'expansion_converter'):
             try:
                 objects_to_expand = self._extract_candidate_objects(actions, expansions)
             except KeyError as e:
                 raise self.InvalidExpansionKey('Invalid key in expansion request: {}'.format(e.args[0]))
             else:
-                self._expand_objects(objects_to_expand)
+                self._expand_objects(objects_to_expand, **kwargs)
 
     def _extract_candidate_objects(self, actions, expansions):
         # Build initial list of objects to expand
         objects_to_expand = []
         for type_node in self.expansion_converter.dict_to_trees(expansions):
             for action in actions:
-                exp_objects = type_node.find_objects(action.body)
+                expansion_objects = type_node.find_objects(action.body)
                 objects_to_expand.extend(
-                    (exp_object, type_node.expansions)
-                    for exp_object in exp_objects
+                    (expansion_object, type_node.expansions)
+                    for expansion_object in expansion_objects
                 )
         return objects_to_expand
 
-    def _expand_objects(self, objects_to_expand):
-            # Keep track of expansion action errors that need to be raised
-            expansion_errors = []
-            # Initialize service request cache
-            exp_service_requests = {}
-            # Loop until we have no outstanding requests or responses
-            while objects_to_expand or any(exp_service_requests.values()):
```

Comment:
I see you removed this extra check for outstanding requests that have not received a response yet. How does your code handle this?

Reply:
That's not a thing that can happen here. `get_all_responses` does just that. It gets _all_ outstanding responses, or it raises a `MessageReceiveTimeout` waiting for any further outstanding responses. `get_all_responses` cannot return only some outstanding requests without raising an exception. So the check for outstanding requests here was unnecessary.